main/src/ca/uwaterloo/flix/api/lsp/provider/completion/CompletionUtils.scala

```diff
@@ -48,6 +48,28 @@ object CompletionUtils {
     (label, Priority.high(s"${exec.getParameterCount}$label"), TextEdit(context.range, replace))
   }
 
+  /**
+    * Used to make a moduleSym from a fqn and the "subWord".
+    *
+    * This function splits the context.word at dots to make the best fqn.
+    * If the word ends at a dot, the subWord is the empty string.
+    *
+    * @param context the completionContext.
+    * @return        a moduleSym from a fqn and a subWord.
+    */
+  def generateModSymAndSubwordFromContext(context: CompletionContext): (Symbol.ModuleSym, String) = {
+    // We use a fqn to generate a modSym
+    // We therefore need to split the context.word at dots
+    val word = context.word.split('.').toList
+
+    // If the word provided ends with a dot, the subWord is the empty string
+    if (context.word.takeRight(1) == ".") {
+      (Symbol.mkModuleSym(word), "")
+    } else {
+      (Symbol.mkModuleSym(word.dropRight(1)), word.takeRight(1)(0))
```

Comment:
I also worry this can go index out of bounds.

Reply:
It won't go out of bounds. It will just return an empty string at worst.

Reply:
Word is a string right?

Reply:
Word is a `List[String]`

Reply:
I believe the last `(0)` can go out of bounds, if the string is empty.

Reply:
Why on earth is "word" a `List[String]` -- thats not good naming.

In any case, if that is the case then `word.takeRight(1)` is `List[String]` and `word.takeRight(1)(0)` can go out of bounds if the list is empty, yes?

Reply:
But `context.word` will never be `Nil`

Reply:
Defensive programming. We cannot know that, if it is a list type it can be `Nil`.

Reply:
> Why on earth is "word" a `List[String]` -- thats not good naming.
> 
> In any case, if that is the case then `word.takeRight(1)` is `List[String]` and `word.takeRight(1)(0)` can go out of bounds if the list is empty, yes?

Bad naming, I've added guards, to ensure that no matter what, we won't ever throw an exception. I'll push a new version with better naming.